lightkurve/lightcurvefile.py

```diff
@@ -148,6 +148,23 @@ class KeplerLightCurveFile(LightCurveFile):
     """
     def __init__(self, path, quality_bitmask='default', **kwargs):
         super(KeplerLightCurveFile, self).__init__(path, **kwargs)
+
+        # check to make sure the correct filetype has been provided
+        filetype = detect_filetype(self.header())
+        if filetype == 'TessLightCurveFile':
+            warnings.warn("A TESS data product is being opened using the "
+                          "`KeplerLightCurveFile` class. "
+                          "Please use `TessLightCurveFile` instead.",
+                          LightkurveWarning)
+        elif filetype is None:
+            warnings.warn("Given fits file not recognized as Kepler or TESS "
+                          "observation.", LightkurveWarning)
+        elif "TargetPixelFile" in filetype:
+            warnings.warn("A `TargetPixelFile` object is being opened as a "
+                          "`KeplerLightCurveFile`. To avoid errors in the "
+                          "future, please use `KeplerTargetPixelFile` instead.",
```

Comment:
We can drop the words `To avoid errors in the future`, because it will almost certainly lead to errors right away!